src/com/ms/silverking/cloud/zookeeper/AsyncCallbackOp.java

```diff
@@ -5,63 +5,65 @@
 import com.ms.silverking.log.Log;
 
 public abstract class AsyncCallbackOp {
-    private final ZKRequestUUID    zkUUID;
-    private State   state;
-    private int     rc;
-    private String  path;
-    
-    private enum State  {incomplete, complete};
-    
-    public AsyncCallbackOp() {
-        zkUUID = new ZKRequestUUID();
-        state = State.incomplete;
-    }
-    
-    public ZKRequestUUID getRequestUUID() {
-        return zkUUID;
-    }
-    
-    public int getRC() {
-        return rc;
-    }
-    
-    public String getPath() {
-        return path;
-    }
-    
-    protected void setCompleteInternal(int rc, String path) {
-        if (state == State.incomplete) {
-            this.rc = rc;
-            this.path = path;
-            state = State.complete;
-            this.notifyAll();
-        } else {
-            Log.warning("Ignoring multiple completions for: ", zkUUID);
-        }
+  private final ZKRequestUUID zkUUID;
+  private State state;
+  private int rc;
+  private String path;
+
+  private enum State {incomplete, complete}
+
+  ;
+
+  public AsyncCallbackOp() {
+    zkUUID = new ZKRequestUUID();
+    state = State.incomplete;
+  }
+
+  public ZKRequestUUID getRequestUUID() {
+    return zkUUID;
+  }
+
+  public int getRC() {
+    return rc;
+  }
+
+  public String getPath() {
+    return path;
+  }
+
+  protected void setCompleteInternal(int rc, String path) {
+    if (state == State.incomplete) {
+      this.rc = rc;
+      this.path = path;
+      state = State.complete;
+      this.notifyAll();
+    } else {
+      Log.warning("Ignoring multiple completions for: ", zkUUID);
     }
-    
-    public void waitForCompletion() {
-        synchronized (this) {
-            while (state == State.incomplete) {
-                try {
-                    this.wait();
-                } catch (InterruptedException ie) {
-                }
-            }
+  }
+
+  public void waitForCompletion() {
+    synchronized (this) {
+      while (state == State.incomplete) {
+        try {
+          this.wait();
+        } catch (InterruptedException ie) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)